app/controllers/users_controller.rb

```diff
@@ -20,9 +20,29 @@ def create
     end
   end
 
+  def edit
+    @user = User.find(current_user.id)
+  end
+
+  def update
+    @user = current_user
+    @user.update(edit_user_params)
+    if @user.save
+      flash[:notice] = "Information Updated"
+      redirect_to "/profile"
+    else
+      flash[:error] = @user.errors.full_messages.to_sentence
+      render :edit
+    end
+  end
+
   private
 
   def user_params
     params.permit(:name, :address, :city, :state, :zip, :email, :password)
   end
+
+  def edit_user_params
```

Comment:
Added `edit_user_params` which does not pass along the user's password when their information is being edited.

Reply:
very good